src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,262 @@
+
+If you have glanced at the C++ example code, you may have noticed the function `ros::spinOnce()` and its differences with `ros::spin()`. Python only has `rospy.spin()`, which can be completely replaced by a for loop, which is why you do not see it in the Python example code.
+
+## Running a Publisher and Subscriber
+Before you are able to run the example code, find the CMakeLists.txt file that is outside of the src directory but still inside of the topic_training directory. This file will include the the configurations, builds, and installs required to run your program as well as the files you will be running. Do not change anything in the file as it should already be set up for the demo files to work. 
+
+The `find_package`, `catkin_package`, and `include_directories` specifications are required for both the C++ and Python files to work. These provide the packages and directories needed for running ROS and ROS topics. 
+
+To run the C++ files, you will need the `add_executable`, `add_dependencies`, and `target_link_libraries` specifications. There will be three of each, representing the three C++ files in this section. `add_executable` determines which file you will use with the command `rosrun`, `add_dependencies` shows what the executable file depends on, and `target_link_libraries` links libraries or executable targets that may be used in the program. In larger projects, these specifications may contain different names, but since the code examples in this training are simple, they are all the same.
+
+For the Python files, you will only need the `catkin_install_python` specification. This will list out the locations of the Python files being used.
+
+In future projects, you might need to use other configurations, builds, and installs, but for the purposes of this training, you only need these for the example code to run.
+
+You can also check out the package.xml file, which you also may need to tailor for future projects. However, everything should already be set up correctly, so do not change anything. At the top of the file, it should name the creator of the package. Then, it will include the license regarding open source policies. Right now, it should not specify a license because the license for the package for the WRoverPlayground training trickles down to this package. At the bottom, there should be various dependencies that allow us to use ROS and ROS topics, similar to the CMakeLists.txt `find_package`.
+
+Now, we will actually run the program.
+
+1. In the terminal, run the command `roscore`. 
+
+2. In a new terminal window, change your current directory to be the catkin workspace. 
+In this case, use `cd WRoverPlayground`. 
+
+3. Use the commands `catkin_make -DCMAKE_EXPORT_COMPILE_COMMANDS=ON` and `source devel/setup.bash` in this order. 
+These commands adjust the catkin workspace based on the CMakeLists.txt and package.xml files. You must run these commands everytime you change these two files. 
```

Comment:
VSCode and the Docker container will take care of these steps for the programmer.  Just use the built-in VSCode terminal

Reply:
Maybe they can stay?  But I think they're redundant.

Reply:
@bennowotny So will students just need to use `rosrun`?

Reply:
Ideally, yes.  The Docker container should take care of the setup automatically.

Reply:
Fixed in 6a86158.